L1 ht

taras-petushkov opened the pull request:
Lesson 1| see commit messages for more details

taras-petushkov committed:
Lesson 1| initial project structure and initial project run

taras-petushkov committed:
Lesson 1| page with all posts

taras-petushkov committed:
Lesson 1| page with specific post

taras-petushkov committed:
Lesson 1| additional task to play with handlebars helpers

taras-petushkov commented on src/post.html:
```diff
@@ -0,0 +1,76 @@
+    <div class="content">
+        <div class="selected-post"></div>
+        <div class="related-posts__list"></div>
+    </div>
```
BEM is something I'm not good in.
I've just added class names similar to (as I saw it) posts.html class names:

```
.content -1> .selected-post -1> .post-preview && .selected-post__post -1> .post-preview__island -1> .post-preview__img-wrapper -1> .post-preview__img
.content -1> .selected-post -1> .post-preview && .selected-post__post -1> .post-preview__island -1> .post-preview__description
.content -1> .selected-post -1> .selected-post__comments -0..n> .selected-post__comment
.content -1> .related-posts__list -0..n> .post-preview && .related-posts__post -1> .post-preview__island -1> .post-preview__img-wrapper -1> .post-preview__img
.content -1> .related-posts__list -0..n> .post-preview && .related-posts__post -1> .post-preview__island -1> .post-preview__island -1> .post-preview__description
```

I think I'll read more about BEM and totally rebuild this page with some modularisation lesson.

puzankov commented on package.json:
```diff
@@ -0,0 +1,12 @@
+        "express": "4.13.3"
+    },
+    "devDependencies": {
+        "bower": "1.7.7"
```
Выполнить задание 3.2
 devDeps

taras-petushkov replied:
done

puzankov commented on package.json:
```diff
@@ -0,0 +1,12 @@
+    "name": "adv-fe",
+    "version": "1.0.0",
+    "description": "More details about the course can be found on http://frontend-science.com/advanced/.",
+    "author": "Petushkov Taras",
```
добавить npm start

taras-petushkov replied:
done

puzankov commented on .bowerrc:
```diff
@@ -0,0 +1,3 @@
+{
+    "directory": "src/lib"
+}
```
update IDE config to add line in the end of file

taras-petushkov replied:
done

puzankov commented on src/post.html:
```diff
@@ -0,0 +1,76 @@
+
+<script type="text/x-handlebars-template" id="related-posts__list-template">
+    {{#each posts}}
+    {{> post-preview class="related-posts__post" }}
```
use tab for child elems

taras-petushkov replied:
done

taras-petushkov committed:
Lesson 1| changes after mentor's review:
+ setup default file ending with eol
+ format handlebars templates: tab on nested elements
+ proper 'any' version dependency for angular: 'x'-->'*'
+ npm start script
+ added missing dev dependencies to package.json: gulp, gulp-clean, run-sequence

puzankov commented:
+1
